src/components/TriggerFolderLink.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { Component, PureComponent } from 'react'
+import classNames from 'classnames'
+
+import Icon from 'cozy-ui/react/Icon'
+
+import styles from '../styles/triggerFolderLink'
+
+const ERRORED = 'ERRORED'
+const FETCHING = 'FETCHING'
+const SUCCESS = 'SUCCESS'
+
+/**
+ * Cache
+ */
+let driveUrl
+
+/**
+ * Renders a link only if href prop is provided
+ * @extends PureComponent
+ */
+class SmartLink extends PureComponent {
+  render() {
+    const { className, href } = this.props
+    return href ? (
+      <a className={className} href={href}>
+        {this.props.children}
+      </a>
+    ) : (
+      <span className={className}>{this.props.children}</span>
+    )
+  }
+}
+
+export class TriggerFolderLink extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
```

Comment:
I would suggest declaring the initial state as a class property, saves us having a constructor.